Sources/Private/CoreAnimation/Animations/CALayer+addAnimation.swift

```diff
@@ -178,11 +178,15 @@ extension CALayer {
       let isLastSegment = (index == animationSegments.indices.last!)
 
       if isFirstSegment {
-        segmentStartTime = context.time(for: context.animation.startFrame)
+        segmentStartTime = min(
+          context.time(for: context.animation.startFrame),
+          segmentStartTime)
       }
 
       if isLastSegment {
-        segmentEndTime = context.time(for: context.animation.endFrame)
+        segmentEndTime = max(
+          context.time(for: context.animation.endFrame),
+          segmentEndTime)
```

Comment:
There was also an issue where incorrect logic here could cause us to create invalid `CAAnimation`s, causing the layer to completely disappear.

In this animation, with a start time of 0 and an end time of 50, had keyframes that spanned from -15 to 65. We were incorrectly truncating the segment start / end times here, which was somehow breaking the entire layer:

| Before | After |
| ---- | ----- |
| ![Simulator Screen Shot - iPhone 8 - 2023-01-06 at 11 46 03](https://user-images.githubusercontent.com/1811727/211089345-623ebedb-3a1e-4cc6-a6bd-01aacd64060e.png) | ![Simulator Screen Shot - iPhone 8 - 2023-01-06 at 11 44 13](https://user-images.githubusercontent.com/1811727/211089355-0bede182-3082-4f99-8bb6-4447f49e26a5.png) |